### ISSUE TITLE
onConnect Hook called twice!

### ISSUE THREAD

**Issue by nshaikhinurov** [Urigo/graphql-modules#380]
I'm using graphql-modules and subscriptions. Some of my modules use a provider with onConnect() method which is supposed to be called once when Apollo Server registers the connection. But onConnect Hook is called twice. And the constructor of my provider is also called twice. I can't understand why this happens.

```js
// SubscriptionsProvider.js

// ...
@Injectable({
	scope: ProviderScope.Session,
})
export class SubscriptionsProvider {
	constructor() {
		this.hash = Math.random()
		console.info('constructor() + hash', this.hash)
	}
        // ...
        onConnect(connectionParams, webSocket, context) {
		console.info('onConnect() + key = ', context.session.socket.upgradeReq.headers['sec-websocket-key'])
	}
}
```

When I subscribe via Apollo playground a resolver gets `SubscriptionsProvider` from `injector` object. And the i see in console that constructor() was called twice with different `hash` part (name is arbitrary) and  onConnect() hook is called twice with the same `sec-websocket-key`

The main goal I wanted to achieve is managing helper websocket connections with my another internal server which provides me with data that i publish to Apollo using onConnect and onDisconnect hooks (to open/close helper websocket connections)

The issue is in surprising additional calls of onConnect and my provider being instantiated twice

**Comment by ardatan** [comment]
@nshaikhinurov Thank you for reporting this bug. I will fix it soon!

**Comment by raulingg** [comment]
In my case this is executed four times, I think that quantity is defined by the quantity of your modules, because i have 4 modules.

**Comment by ardatan** [comment]
It is available in 0.7.0.
